src/main/java/io/sealights/plugins/sealightsjenkins/BeginAnalysis.java

```diff
@@ -572,43 +572,24 @@ private String joinPaths(String path1, String path2) {
         }
         return Paths.get(path1, path2).toAbsolutePath().toString();
     }
-
-    private String getManualBuildName() {
-        BuildName.ManualBuildName manual = (BuildName.ManualBuildName) buildName;
-        String insertedBuildName = manual.getInsertedBuildName();
-        return insertedBuildName;
-    }
-
-    private String getFinalBuildName(AbstractBuild<?, ?> build, SeaLightsPluginInfo slInfo, Logger logger) throws IllegalStateException {
-
+    
+    private String getFinalBuildName(AbstractBuild<?, ?> build, Properties additionalProps, Logger logger) throws IllegalStateException {
         String finalBuildName = null;
+        if (additionalProps.getProperty("buildname") != null) {
+            finalBuildName = additionalProps.getProperty("buildname").toString();
```

Comment:
Why do you need to call 'toStirng()'? Seems like 'getProperty()' is returning string.